src/policyengine_data/single_year_dataset.py

```diff
@@ -1 +1,157 @@
+        if not Path(file_path).exists():
+            raise FileNotFoundError(f"File not found: {file_path}")
+
+        with h5py.File(file_path, "r") as f:
+            required_datasets = [
+                "time_period",
+                "person",
+                "household",
+            ]  # all datasets will have at least person and household entities
+            for dataset in required_datasets:
+                if dataset not in f:
+                    raise ValueError(
+                        f"Dataset '{dataset}' not found in the file: {file_path}"
+                    )
+
+    def save(self, file_path: str) -> None:
+        with pd.HDFStore(file_path) as f:
+            for entity, df in self.entities.items():
+                f.put(entity, df, format="table", data_columns=True)
+            f.put("time_period", pd.Series([self.time_period]), format="table")
+
+    def load(self) -> Dict[str, pd.Series]:
+        data = {}
+        for entity_name, entity_df in self.entities.items():
+            for col in entity_df.columns:
+                data[col] = entity_df[col].values
+
+        return data
+
+    def copy(self) -> "SingleYearDataset":
+        return SingleYearDataset(
+            entities={name: df.copy() for name, df in self.entities.items()},
+            fiscal_year=self.time_period,
+        )
+
+    def remove(self) -> None:
+        if hasattr(self, "file_path") and self.file_path is not None:
+            file_path = (
+                Path(self.file_path)
+                if isinstance(self.file_path, str)
+                else self.file_path
+            )
+            if file_path.exists():
+                file_path.unlink()
+                print(f"Removed dataset file: {file_path}")
+            else:
+                print(f"Dataset file does not exist: {file_path}")
+        else:
+            raise FileNotFoundError(
+                "Cannot remove dataset: no file path associated with this dataset. "
+                "This dataset may have been created in-memory."
+            )
+
+    def validate(self) -> None:
+        # Check for NaNs in the tables
+        for df in self.tables:
+            for col in df.columns:
+                if df[col].isna().any():
+                    raise ValueError(f"Column '{col}' contains NaN values.")
+
+    def variables(self) -> Dict[str, List[str]]:
+        """
+        Returns a dictionary mapping entity names to lists of variables (column names).
+
+        Returns:
+            Dict[str, List[str]]: Dictionary where keys are entity names and values are lists of variable names for that entity.
+        """
+        variables_by_entity = {}
+
+        for entity_name, entity_df in self.entities.items():
+            variables_by_entity[entity_name] = entity_df.columns.tolist()
+
+        return variables_by_entity
+
+    @staticmethod
+    def from_simulation(
+        simulation: "Microsimulation",
+        fiscal_year: int = 2025,
```

Comment:
I guess we should probably make this `time_period` everywhere to avoid confusion in retrospect.

I despise the constant swapping between ints and strings and we should change Core to accept ints everywhere. but fine for now to have this two-tier system.